tests/manage/mcg/test_noobaa_secret.py

```diff
@@ -0,0 +1,332 @@
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.framework import config
+from ocs_ci.ocs.exceptions import CommandFailed
+from ocs_ci.utility.aws import update_config_from_s3
+from ocs_ci.utility.utils import load_auth_config
+from botocore.exceptions import EndpointConnectionError
+
+logger = logging.getLogger(__name__)
+
+
+def create_bs_using_cli(
+    mcg_obj, access_key, secret_key, backingstore_name, uls_name, region
+):
+
+    mcg_obj.exec_mcg_cmd(
+        f"backingstore create aws-s3 {backingstore_name} "
+        f"--access-key {access_key} "
+        f"--secret-key {secret_key} "
+        f"--target-bucket {uls_name} --region {region}",
+        use_yes=True,
+    )
+
+
+@pytest.fixture(scope="function")
+def cleanup(request):
+    """
+    Clean up function for the backinstores created using CLI
+    where we can't use teardown_factory()
+
+    """
+    instances = []
+
+    def factory(resource_obj):
+
+        if isinstance(resource_obj, list):
+            instances.extend(resource_obj)
+        else:
+            instances.append(resource_obj)
+
+    def finalizer():
+        for instance in instances[::-1]:
+            try:
+                instance.delete()
+            except CommandFailed as ex:
+                if "not found" in str(ex).lower():
+                    logger.warning(
+                        f"Resource {instance.name} could not be found in cleanup."
+                        "\nSkipping deletion."
+                    )
+                else:
+                    raise
+
+    request.addfinalizer(finalizer)
+    return factory
+
+
+class TestNoobaaSecrets:
+    def test_duplicate_noobaa_secrets(
+        self, backingstore_factory, cloud_uls_factory, mcg_obj, teardown_factory
+    ):
+        """
+        Objective of this test is:
+            1) Create a secret with the same credentials and see if the duplicates are allowed when BS created
+            2) Modify the existing secret credentials see if the owned BS/NS is getting reconciled
+
+        """
+        # create secret with the same credentials to check if duplicates are allowed
+        first_bs_obj = backingstore_factory(
+            method="oc", uls_dict={"aws": [(1, "eu-central-1")]}
+        )[0]
+        try:
+            logger.info(
+                "Trying to load credentials from ocs-ci-data. "
+                "This flow is only relevant when running under OCS-QE environments."
+            )
+            secret_dict = update_config_from_s3().get("AUTH")
+        except (AttributeError, EndpointConnectionError):
+            logger.warning(
+                "Failed to load credentials from ocs-ci-data.\n"
+                "Your local AWS credentials might be misconfigured.\n"
+                "Trying to load credentials from local auth.yaml instead"
+            )
+            secret_dict = load_auth_config().get("AUTH", {})
+        aws_client = S3Client(auth_dict=secret_dict["AWS"])
+        teardown_factory(aws_client.secret)
```

Comment:
First of all, `cld_mgr.aws_client.secret_prefix` and `cld_mgr.aws_client.data_prefix` doesn't exists because we are not initializing aws_client as per your suggestion. And this will lead us to have `secret_prefix` and `data_prefix` fetched prior. so the code will look something like this:
`        try:
            logger.info(
                "Trying to load credentials from ocs-ci-data. "
                "This flow is only relevant when running under OCS-QE environments."
            )
            secret_dict = update_config_from_s3().get("AUTH")
        except (AttributeError, EndpointConnectionError):
            logger.warning(
                "Failed to load credentials from ocs-ci-data.\n"
                "Your local AWS credentials might be misconfigured.\n"
                "Trying to load credentials from local auth.yaml instead"
            )
            secret_dict = load_auth_config().get("AUTH", {})
            secret_prefix = secret_dict["AWS"].get("SECRET_PREFIX", "AWS")
        data_prefix = secret_dict["AWS"].get("DATA_PREFIX", "AWS")
        aws_secret_obj = cld_mgr.aws_client.create_s3_secret(secret_prefix, data_prefix)
        teardown_factory(aws_secret_obj)
`
which do you think is better? @MeridianExplorer

Reply:
I will probably use the second method because instantiating an object seems to be expensive!

Reply:
@MeridianExplorer 
secret_prefix is not an instance variable, IIUC
> class S3Client(CloudClient):
    """
    Implementation of a S3 Client using the S3 API

    """

    def __init__(
        self,
        auth_dict,
        verify=True,
        endpoint="https://s3.amazonaws.com",
        *args,
        **kwargs,
    ):
        super().__init__(*args, **kwargs)

        secret_prefix = auth_dict.get("SECRET_PREFIX", "AWS")
        data_prefix = auth_dict.get("DATA_PREFIX", "AWS")
        key_id = auth_dict.get(f"{secret_prefix}_ACCESS_KEY_ID")
        access_key = auth_dict.get(f"{secret_prefix}_SECRET_ACCESS_KEY")
        self.endpoint = auth_dict.get("ENDPOINT") or endpoint
        self.region = auth_dict.get("REGION")
        self.access_key = key_id
        self.secret_key = access_key